src/hierarchical-menu/hierarchical-menu.ts

```diff
@@ -5,10 +5,25 @@ import { BaseWidget } from '../base-widget';
 import { NgAisInstantSearch } from '../instantsearch/instantsearch';
 import { parseNumberInput, noop } from '../utils';
 
+export type HierarchicalMenuSortByStringOptions =
+  | 'count:asc'
+  | 'count:desc'
+  | 'name:asc'
+  | 'name:desc'
+  | 'isRefined';
+
 export type HierarchicalMenuState = {
-  createURL: Function;
-  items: {}[];
-  refine: Function;
+  createURL: (value: string) => string;
+  items: HierarchicalMenuItem[];
+  refine: (value: string) => void;
```

Comment:
The widget also supports the "Show more" behaviour. We don't have to tackle it right now though.

Reply:
Good catch. Not vital to releasing a stable I think, so I opened an issue for it.